src/librustc/infer/error_reporting/util.rs

```diff
@@ -79,22 +106,26 @@ impl<'a, 'gcx, 'tcx> InferCtxt<'a, 'gcx, 'tcx> {
 
     // This method returns whether the given Region is Anonymous
     // and returns the DefId and the BoundRegion corresponding to the given region.
+    // The is_anon_anon is set true when we are dealing with cases where
```

Comment:
Could you *please* store `is_impl_item` in `FreeRegionInfo` and branch on it instead of passing `is_anon_anon` as a parameter? It being a parameter is confusing.